internal/server/server.go

```diff
@@ -79,6 +85,15 @@ func (srv *Server) Provision(_ context.Context, req *pb.ProvisionRequest) (*pb.P
 		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("error parsing Wireguard key: %s", err))
 	}
 
+	var grpcPeerAddrPort string
+
+	if req.WireguardOverGrpc != nil && *req.WireguardOverGrpc {
+		grpcPeerAddrPort, err = srv.cfg.PeerProvider.PeerAddrPort(pubKey)
```

Comment:
if the tunnel is requested, the peer address over the tunnel should be generated as above, when the `nodeAdress` is generated from the prefix (this code needs a prefix to allocate from)

Reply:
Done!